test/test.js

```diff
@@ -236,4 +236,39 @@ describe('Chinese calendar', function() {
                 ).toEqual(testCase.chinese);
         });
     });
+
+    it('should compute the number of months in a year correctly', function() {
+        for(var year = 1888; year < 2112; year++) {
+            var monthsInYear = (chineseCalendar.leapYear(year)) ? 13 : 12;
+            expect(chineseCalendar.monthsInYear(year)).toEqual(monthsInYear);
+        }
+    });
+
+    it('should keep months in sync when adding years', function() {
+        testCases.forEach(function(testCase) {
+            var chineseDate =
+                chineseCalendar.parseDate(testCase.format, testCase.chinese);
+
+            var year = chineseDate.year();
+            var monthIndex = chineseDate.month();
+            var month = chineseCalendar.toChineseMonth(year, monthIndex);
+            var isIntercalary =
+                chineseCalendar.intercalaryMonth(year, monthIndex);
+
+            chineseDate.add(1, 'y');
+
+            var resultYear = chineseDate.year();
+            var resultLeapYear = chineseDate.leapYear();
+            var resultMonthIndex = chineseDate.month();
+            var resultMonth =
+                chineseCalendar.toChineseMonth(resultYear, resultMonthIndex);
+            var resultIsIntercalary =
+                chineseCalendar.intercalaryMonth(resultYear, resultMonthIndex);
+
+            expect(resultYear).toEqual(year + 1);
+            expect(resultMonth).toEqual(month);
+            expect(resultIsIntercalary)
+                .toEqual(isIntercalary && resultLeapYear);
```

Comment:
you're not hitting this case, are you? because you're only adding one year in the test, so you'll never have both the start and end year be leap years. But if you did I don't think this is the right test - you should only increment from intercalary to intercalary if it's the *same* intercalary month in the end year.